storage-access.bs

```diff
@@ -232,53 +197,53 @@ ISSUE: Shouldn't step 9 be [=same site=]?
-
-    Note: While |expressly granted| and |expressly denied| cannot both be true, they could both be false in User Agents which allow users to dismiss the prompt without choosing to allow or deny the request. (Such a dismissal is interpreted in this algorithm as a denial.)
-1. If |expressly granted| is true, run these steps:
-    1. Unset |flag set|'s [=was expressly denied storage access flag=].
-    1. [=Save the storage access flag set=] for |key| in |map|.
-    1. [=Queue a global task=] on the [=permission task source=] given |global| to [=/resolve=] |p|, and return.
-1. Unset |flag set|'s [=has storage access flag=].
-1. If |expressly denied| is true, run these steps:
-    1. If |doc|'s {{Window}} object has [=transient activation=], [=consume user activation=] with it.
-    1. Set |flag set|'s [=was expressly denied storage access flag=].
-1. [=Save the storage access flag set=] for |key| in |map|.
+1. If |global|'s [=environment/has storage access=] is true, [=resolve=] |p| with {{undefined}} and return.
+1. Let |previousPermissionState| be the result of [=getting the current permission state=] given "<a permission><code>storage-access</code></a>" and |global|.
+1. If |previousPermissionState| is not [=permission/prompt=]:
+    1. If |previousPermissionState| is [=permission/granted=]:
+        1. Set |global|'s [=environment/has storage access=] to true.
+        1. [=/Resolve=] |p| with {{undefined}}.
+        1. Return.
+    1. [=/Reject=] |p| with a "{{NotAllowedError}}" {{DOMException}}.
+    1. Return.
+1. If |doc|'s {{Window}} object does not have [=transient activation=], [=reject=] |p| with a "{{NotAllowedError}}" {{DOMException}} and return.
+1. Let |key| be the result of [=generate a partitioned storage key|generating a partitioned storage key=] from |doc|.
+1. Let |implicitly granted| and |implicitly denied| (each a [=boolean=]) be the result of running an [=implementation-defined=] set of steps to determine if |key|'s [=partitioned storage key/embedded origin=]'s request for storage access on |key|'s [=partitioned storage key/top-level site=] should be granted or denied without prompting the user.
+1. If |implicitly granted| is true, [=queue a global task=] on the [=permission task source=] given |global| to [=/resolve=] |p| with {{undefined}}, and return.
+1. If |implicitly denied| is true, [=queue a global task=] on the [=permission task source=] given |global| to [=/reject=] |p| with a "{{NotAllowedError}}" {{DOMException}}, and return.
+1. Let |permissionState| be the result of [=requesting permission to use=] "<a permission><code>storage-access</code></a>".
+1. If |permissionState| is "granted":
+    1. [=Queue a global task=] on the [=permission task source=] given |global| to:
+        1. Set |global|'s [=environment/has storage access=] to true.
+        1. [=/Resolve=] |p| with {{undefined}}.
+    1. Return.
+1. [=Consume user activation=] given |doc|'s {{Window}} object.
```

Comment:
Note @annevk's comment here: https://github.com/privacycg/storage-access/pull/138#discussion_r1047339352

Which I think still applies to this PR

Reply:
Done (I think).